pandas/core/generic.py

```diff
@@ -6139,12 +6139,12 @@ def convert_dtypes(
         2  3  z   <NA>  <NA>    20  200.0
 
         >>> dfn.dtypes
-        a      Int32
-        b     string
-        c    boolean
-        d     string
-        e      Int64
-        f    Float64
+        a              Int32
+        b     string[python]
```

Comment:
made the changes. will commit shortly once more thoroughly tested.

Reply:
this could make debugging a bit harder for string dtypes, may need to special case the asserters in a follow-up

```
       AssertionError: Attributes of DataFrame.iloc[:, 0] (column name="0") are different
E       
E       Attribute "dtype" are different
E       [left]:  string
E       [right]: string
```

Reply:
also raises questions about out usage of `str()` and `repr()`, we should maybe not be changing `repr()` to allow reconstruction of the object and not have a test in the base extension tests using `str()` in test_from_dtype

Reply:
> this could make debugging a bit harder for string dtypes, may need to special case the asserters in a follow-up

there is already some special casing there, so I think OK to change in this PR

Reply:
> not have a test in the base extension tests using `str()` in test_from_dtype

changing to `repr()` and tests fail for other dtypes, will need to skip that test for StringDtype

Reply:
Yes, I think it's fine that we would start using `repr` instead of `str` in the assert output (and it's fine to only do that for string dtype in this PR, as you did now, we can see if we want to generalize that later)